src/app/pages/AccountPage/Features/TransactionHistory/index.tsx

```diff
@@ -11,39 +11,95 @@ import { useTranslation } from 'react-i18next'
   const { t } = useTranslation()
   const allTransactions = useSelector(selectTransactions)
   const transactionsError = useSelector(selectTransactionsError)
   const address = useSelector(selectAccountAddress)
+  const pendingTransactions = useSelector(selectPendingTransactionForAccount)
+  const hasUnknownPendingTransactions = useSelector(hasAccountUnknownPendingTransactions)
   const network = useSelector(selectSelectedNetwork)
-  const transactionComponents = allTransactions.map((t, i) => (
-    <Transaction key={i} transaction={t} referenceAddress={address} network={network} />
+  const backendLinks = config[network][backend()]
+  const transactionComponents = allTransactions.map(t => (
+    <Transaction key={t.hash} transaction={t} referenceAddress={address} network={network} />
   ))
+  const pendingTransactionComponents = pendingTransactions
+    .filter(({ hash: pendingTxHash }) => !allTransactions.some(({ hash }) => hash === pendingTxHash))
+    .map(t => <Transaction key={t.hash} transaction={t} referenceAddress={address} network={network} />)
 
   return (
-    <Box gap="medium" margin="none">
+    <Box margin="none">
       {transactionsError && (
         <p>
-          {t('account.transaction.loadingError', "Couldn't load transactions.")}{' '}
+          {t('account.transaction.loadingError', `Couldn't load transactions.`)}{' '}
           <ErrorFormatter code={transactionsError.code} message={transactionsError.message} />
         </p>
       )}
+      {/* eslint-disable no-restricted-syntax */}
+      {(!!pendingTransactionComponents.length || hasUnknownPendingTransactions) && (
+        <>
+          <Heading level="3">{t('account.summary.pendingTransactions', 'Pending transactions')}</Heading>
+          <AlertBox
+            status="ok-weak"
+            justify="between"
+            endSlot={
+              <span>
+                {backendLinks.blockExplorerAccount && (
+                  <Anchor
+                    href={backendLinks.blockExplorerAccount.replace(
+                      '{{address}}',
+                      encodeURIComponent(address),
+                    )}
+                    target="_blank"
+                    rel="noopener"
+                    color="brand"
+                    style={{ verticalAlign: 'middle' }}
+                  >
+                    <Text size="xsmall">
+                      {t('account.summary.viewAccountTxs', 'View account transactions')}
+                    </Text>
+                    <FormNext color="brand" size="20px" />
+                  </Anchor>
+                )}
+              </span>
+            }
+          >
+            {t(
+              'account.summary.someTxsInPendingState',
+              'Some transactions are currently in a pending state.',
+            )}
+          </AlertBox>
+          {pendingTransactionComponents.length && (
+            <Box gap="medium" data-testid="pending-txs" margin={{ top: 'small' }}>
```

Comment:
This prints "0" when hasUnknownPendingTransactions
![image](https://github.com/oasisprotocol/oasis-wallet-web/assets/3758846/07224207-d9e1-4161-807a-1f08f84991e3)

Related https://github.com/oasisprotocol/explorer/issues/259

```suggestion
          {pendingTransactionComponents.length > 0 && (
            <Box gap="medium" data-testid="pending-txs" margin={{ top: 'small' }}>
```